### DISCUSSION TITLE
help

### DISCUSSION THREAD

**Discussion by sitharah** [L3MON4D3/LuaSnip#505] [Q&A]
Hi lua devs, I need your help . I installed Luasnip,friendly-snippets and cmp_lua for auto completion. Now when I type the suggestions shows up but I don't know how to use them. don;t know about the keybindings. Please if you can help me to make keybindings like:
I only know the very basics of lua.
keymaps want to use:
j for the next snippet and k for the previous snippet. TAB to show and hide snippets and Enter to select the snippet
Thanks...

**Comment by L3MON4D3** [comment]
Hi :)

I think you'll want those mappings for `nvim-cmp`, the completion-engine, right?
Take a look in `:h cmp-mapping`, it shows how to map the different actions, short example:
```lua
cmp.setup({
  mapping = {
    ["<C-j>"] = cmp.mapping.select_next_item,
    ...
  }
})
```

Using one key to toggle visibility of `nvim-cmp` like you want to do with `<TAB>` isn't possible out of the box I think, so you'll need to write your own function for that.
